### PR TITLE
Disposal drop their contents if they're destroyed or unanchored. Fixes #4114 

### LINKED ISSUE #4114: Disposal outlets don't drop their contents if they're destroyed while unanchored.
This was originally reported by Seþ#2464. I have confirmed this is reproducible in a fully-updated dev environment.

How to reproduce:

- Put someone in a disposals outlet.
- Unanchor the outlet.
- Destroy the unanchored outlet. The outlet will drop some steel sheets but will not drop its contents.

Ideally, outlets should drop their contents when unanchored (and prevent anything else from going in when unanchored) to prevent this.

### PR DESCRIPTION
**Pull request by Fiftyllama**
Disposals now drop their contents if destroyed or if unwrenched

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
Nice first PR.

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs [hunk 1]
```diff
@@ -9,15 +10,22 @@ public override void Initialize()
         {
             base.Initialize();
 
-            SubscribeLocalEvent<DisposalUnitComponent, PhysicsBodyTypeChangedEvent>(BodyTypeChanged);
+
+            SubscribeLocalEvent<DisposalUnitComponent, AnchoredEvent>(OnAnchored);
+            SubscribeLocalEvent<DisposalUnitComponent, UnanchoredEvent>(OnUnanchored);
+
+
+
```
Comment (mirrorcult) [inline Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:18]: remove this extra whitespace

**Review by mirrorcult — APPROVED** [review]
good first PR

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs (hunk 1 above)
Comment (Visne) [reply Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:18]: And readd the newline after the closing bracket.

Comment (Fiftyllama) [reply Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs]: Like so?

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs [hunk 2]
```diff
@@ -9,15 +10,20 @@ public override void Initialize()
         {
             base.Initialize();
 
-            SubscribeLocalEvent<DisposalUnitComponent, PhysicsBodyTypeChangedEvent>(BodyTypeChanged);
+
```
Comment (ShadowCommander) [inline Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:13]: Remove extra newline

**Review by ShadowCommander — APPROVED** [review]

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs (hunk 2 above)
Comment (Fiftyllama) [reply Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:13]: i think i've done that one too, this time without removing the ; while im at it...
hopefully

**Pull request merged**